ic-transport-types/src/request_id.rs

```diff
@@ -854,20 +652,154 @@ mod tests {
         */
     }
 
-    /// We do not support creating a request id from a map.
-    /// It adds complexity, and isn't that useful anyway because a real request would
-    /// have to have different kinds of values (strings, principals, arrays) and
-    /// we don't support the wrappers that would be required to make that work
-    /// with rust maps.
     #[test]
-    fn maps_are_not_supported() {
-        let mut data = BTreeMap::new();
-        data.insert("request_type", "call");
-        data.insert("canister_id", "a principal / the canister id");
-        data.insert("method_name", "hello");
-        data.insert("arg", "some argument value");
-
-        let error = to_request_id(&data).unwrap_err();
-        assert_eq!(error, RequestIdError::UnsupportedTypeMap);
+    fn nested_map() {
+        #[derive(Serialize)]
+        struct Outer {
+            foo: Inner,
+            #[serde(with = "serde_bytes")]
+            bar: &'static [u8],
+        }
+        #[derive(Serialize)]
+        struct Inner {
+            baz: &'static str,
+            quux: u64,
+        }
+        let outer = Outer {
+            foo: Inner {
+                baz: "hello",
+                quux: 3,
+            },
+            bar: b"world",
+        };
+        assert_eq!(
+            hex::encode(to_request_id(&outer).unwrap().0),
+            "3d447339cc0c2b894ee215c8141770bf4b86c72b6c37d9873213a786ec7f9f31"
```

Comment:
Could we add tests like this (compare computed hash with specific expected hash) for Tuple, Newtype, and any other previously unsupported types?